Lib/pathlib.py

```diff
@@ -86,6 +86,12 @@ def _make_selector(pattern_parts, flavour, case_sensitive):
     return cls(pat, child_parts, flavour, case_sensitive)
 
 
+@functools.lru_cache(maxsize=256)
+def _compile_pattern(pat, case_sensitive):
```

Comment:
Last thing: could you call this function from `_WildcardSelector.__init__()`?

The difference between `match` and `fullmatch` shouldn't matter, as `fnmatch.translate()` results in a pattern ending with `\Z`.